public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -526,9 +529,14 @@ const getStyles = (theme: GrafanaTheme2) => ({
 
 const useSetExpressionAndDataSource = () => {
   const { setValue } = useFormContext<RuleFormValues>();
+
   return (updatedQueries: AlertQuery[]) => {
     // update data source name and expression if it's been changed in the queries from the reducer when prom or loki query
     const query = updatedQueries[0];
+    if (!query) {
```

Comment:
This prevents the UI from crashing since we can't assume `updatedQueries[0]` to not be `[]` when the last data query is removed.